domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -261,37 +340,343 @@ class TopicListController @Inject constructor(
+    )
+
+  private fun getCompletedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> =
+    getSortedChapterProgressListByPlayState(
+      storyProgress, playState = ChapterPlayState.COMPLETED
+    )
+
+  private fun getSortedChapterProgressListByPlayState(
+    storyProgress: StoryProgress,
+    playState: ChapterPlayState
+  ): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress -> chapterProgress.chapterPlayState == playState }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun createOngoingStoryListBasedOnRecentlyPlayed(
+    storyId: String,
+    story: StorySummary,
+    latestStartedChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    isTopicConsideredCompleted: Boolean
+  ): PromotedStory? {
+    val recentlyPlayerChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        latestStartedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    if (recentlyPlayerChapterSummary != null) {
+      return createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        recentlyPlayerChapterSummary.name,
+        recentlyPlayerChapterSummary.explorationId,
+        isTopicConsideredCompleted
+      )
+    }
+    return null
+  }
+
+  private fun createOngoingStoryListBasedOnMostRecentlyCompleted(
+    storyId: String,
+    story: StorySummary,
+    latestCompletedChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    isTopicConsideredCompleted: Boolean
+  ): PromotedStory? {
+    val lastChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        latestCompletedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
+    if (story.chapterList.size > nextChapterIndex) {
+      val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+      if (nextChapterSummary != null) {
+        return createPromotedStory(
+          storyId,
+          topic,
+          completedChapterProgressList.size,
+          story.chapterCount,
+          nextChapterSummary.name,
+          nextChapterSummary.explorationId,
+          isTopicConsideredCompleted
+        )
+      }
+    }
+    return null
+  }
+
+  private fun ChapterProgress.getNumberOfDaysPassed(): Long {
+    return TimeUnit.MILLISECONDS.toDays(
+      oppiaClock.getCurrentCalendar().timeInMillis - this.lastPlayedTimestamp
+    )
+  }
+
+  // TODO(#2550): Remove hardcoded order of topics. Compute list of suggested stories from backend structures
+  /** Returns a list of topic IDs for which the specified topic ID expects to be completed before
+   * being suggested.
+   */
```

Comment:
```suggestion
  /**
   * Returns a list of topic IDs for which the specified topic ID expects to be completed before
   * being suggested.
   */
```

Reply:
NB: I think you missed the earlier version of this comment, but I did it again in the latest review to make it easier to find.